### PR TITLE
update recipe per lint's suggestion

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
use build/compiler, host with pin_compatible for numpy per lint's suggestion.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -14,16 +14,19 @@ build:
 
 requirements:
     build:
+      - compiler('c')
```
Comment (isuruf) [inline recipe/meta.yaml]: This needs to be `{{ compiler('c') }}`

Comment (isuruf) [reply recipe/meta.yaml]: `- {{ compiler('c') }}`

Comment (oleksandr-pavlyk) [reply recipe/meta.yaml]: @isuruf Thank you. I missed that in the rush. Committed the fix.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -14,16 +14,17 @@ build:
...
     build:
+      - {{ compiler('c') }}
+    host:
       - python
-      - toolchain
       - setuptools
       - mkl-devel
       - cython
-      - numpy 1.11.*
+      - numpy
```
Comment (xhochy) [inline recipe/meta.yaml:23]: Restore 1.11.* here

File: recipe/meta.yaml [hunk 3]
```diff
@@ -14,16 +14,17 @@ build:
...
     run:
       - python
       - mkl
-      - numpy >=1.11
+      - {{ pin_compatible('numpy') }}
```
Comment (xhochy) [inline recipe/meta.yaml:27]: Add `min_pin='1.11.x'

**Review by xhochy — COMMENTED** [review]
This fails as the minimal supported NumPy version is 1.11 but it tries to build with 1.9

File: recipe/meta.yaml (hunk 3 above)
Comment (isuruf) [reply recipe/meta.yaml:27]: That's unnecessary

**Comment by jschueller** [comment]
This PR can be closed, the meta.yaml already has the new compiler syntax.

**Comment by oleksandr-pavlyk** [comment]
Equivalent changes are already in master.

**Pull request closed without merging**